indico/modules/categories/controllers/management.py

```diff
@@ -16,10 +16,44 @@
 
 from __future__ import unicode_literals
 
+from flask import redirect
+
+from indico.web.forms.base import FormDefaults
 from indico.modules.categories.controllers.base import RHManageCategoryBase
+from indico.modules.categories.forms import CategoryProtectionForm
+from indico.modules.categories.operations import update_category
 from indico.modules.categories.views import WPCategoryManagement
+from indico.modules.events.util import update_object_principals
+from indico.web.flask.util import url_for
 
 
 class RHManageCategoryContent(RHManageCategoryBase):
     def _process(self):
         return WPCategoryManagement.render_template('management/content.html', self.category, 'content')
+
+
+class RHManageCategoryProtection(RHManageCategoryBase):
+    def _process(self):
+        form = CategoryProtectionForm(obj=FormDefaults(**self._get_defaults()), category=self.category)
+        if form.validate_on_submit():
+            update_category(self.category,
+                            {'protection_mode': form.protection_mode.data,
+                             'no_access_contact': form.no_access_contact.data,
+                             'event_creation_restricted': form.event_creation_restricted.data,
+                             'event_creation_notification_emails': form.event_creation_notification_emails.data})
+            update_object_principals(self.category, form.acl.data, read_access=True)
+            update_object_principals(self.category, form.managers.data, full_access=True)
+            update_object_principals(self.category, form.event_creators.data, role='create')
+            return redirect(url_for('.manage_protection', self.category))
+        return WPCategoryManagement.render_template('management/category_protection.html', self.category, 'protection',
+                                                    form=form)
+
+    def _get_defaults(self):
+        acl = {x.principal for x in self.category.acl_entries if x.read_access}
+        managers = {x.principal for x in self.category.acl_entries if x.full_access}
+        event_creators = {x.principal for x in self.category.acl_entries
+                          if x.has_management_role('create', explicit=True)}
+        return {'protection_mode': self.category.protection_mode, 'acl': acl, 'managers': managers,
```

Comment:
Maybe see if `FormDefaults(self.category, acl=acl, managers=managers, ...)` works. Avoids repeating the "simple" fields. Probably you need to add acl, managers, etc to `skip_attrs` as FormDefaults prefers object attributes over kwargs.